src/sentry/web/frontend/auth_channel_login.py

```diff
@@ -1,18 +1,37 @@
 from django.urls import reverse
 from django.views.decorators.cache import never_cache
 
+from sentry.auth.helper import CHANNEL_PROVIDER_MAP
+from sentry.models.authprovider import AuthProvider
 from sentry.models.organization import Organization
 from sentry.services.hybrid_cloud.organization import organization_service
 from sentry.utils.auth import is_valid_redirect
 from sentry.web.frontend.auth_organization_login import AuthOrganizationLoginView
 
 
-class AuthOrganizationIdentifierLoginView(AuthOrganizationLoginView):
+class AuthChannelLoginView(AuthOrganizationLoginView):
     @never_cache
-    def handle(self, request, organization_id):
+    def handle(self, request, channel):
         if request.subdomain is not None:
             return self.redirect(reverse("sentry-auth-organization", args=[request.subdomain]))
 
+        config_data = request.GET.get("config_data", "")
+        resource_id = request.GET.get("organization_id", "")
```

Comment:
Do we want a serializer to handle this? I'm unfamiliar with views so this might be completely wrong

Reply:
Also, should we call the query parameter `resource_id` instead of `organization_id` to avoid confusing it with the sentry org id?

Reply:
ah yes - 

except this is already a subclass of a subclass of a class that is far too complex (also not serialized) - so rather than attempting to rebuild this tree, its easier to just keep it as is 😬